tests/manage/pv_services/test_overprovision_level_policy_control_with_capacity.py

```diff
@@ -0,0 +1,208 @@
+                status=constants.STATUS_BOUND,
+            )
+        except Exception as e:
+            log.error(f"Failed to create PVC {str(e)}")
+            assert False
+
+        clusterresourcequota_obj = OCP(kind="clusterresourcequota")
+        output_clusterresourcequota = clusterresourcequota_obj.describe(
+            resource_name=constants.CEPHBLOCKPOOL_SC
+        )
+
+        log.info(f"Output Cluster Resource Quota: {output_clusterresourcequota}")
+
+        assert self.verify_substrings_in_string(
+            output_string=output_clusterresourcequota,
+            expected_strings=["50Gi"])
+
+
+        log.info(
+            "Add another pvc with 51Gi capacity and verify it failed [50Gi + 51Gi > 100Gi]"
+        )
+        try:
+            pvc_factory(
+                interface=sc_type,
+                project=ocp_project_obj,
+                storageclass=sc_obj,
+                size=51,
+            )
+        except Exception as e:
+            assert self.verify_substrings_in_string(
+                output_string=str(e), expected_strings=["forbidden","exceeded quota"]
+            ), f"The error does not contain string:{str(e)}"
+
+        log.info("Removing overprovisionControl from storage cluster.")
+        storagecluster_obj = OCP(
+            resource_name=constants.DEFAULT_CLUSTERNAME,
+            namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+            kind=constants.STORAGECLUSTER,
+        )
+        params = '{"spec": {"overprovisionControl": []}}'
+        storagecluster_obj.patch(
+            params=params,
+            format_type="merge",
+        )
+        log.info("Verify storagecluster on Ready state.")
+        verify_storage_cluster()
+
+        """ Adding the new PVC with 51Gi  """
+        try:
+            pvc_factory(
+                interface=sc_type,
+                project=ocp_project_obj,
+                storageclass=sc_obj,
+                size=51,
+            )
+        except Exception as e:
+            log.error(f"Failed to create PVC : {e}")
+            assert False
+
+        output_clusterresourcequota = clusterresourcequota_obj.describe(
+            resource_name=constants.CEPHBLOCKPOOL_SC
+        )
+        log.info(f"Output Cluster Resource Quota: {output_clusterresourcequota}")
+
+        assert self.verify_substrings_in_string(
+            output_string=output_clusterresourcequota,
+            expected_strings=["50Gi","51Gi"])
+
+    def verify_substrings_in_string(self, output_string, expected_strings):
+        """
+        Verify substrings in string
+
+        Args:
+           output_string (str): the output of cmd
+           expected_strings (list) : list of strings
+
+        Returns:
+            bool: return True if all expected_strings in output_string, otherwise False
+
+        """
+        if output_string is None:
+            return False
+
+        for expected_string in expected_strings:
+            if expected_string not in output_string:
```

Comment:
You are assuming that the output string is a list, but then in the first false you are returning false so if the first line doesn't contain the output you are searching, the loop quits because of the return. Maybe do it the other way like
if expected_string in output_string:
 return true
And in the end of the loop return false.

Reply:
I have modified the logic to this way. I kept one list which will store  all the results of match string iterations . And on  the basis of that result  code is  returning True or false. If any one false value present in the list it will always return False and only if all values are true in the list then only it will return True.